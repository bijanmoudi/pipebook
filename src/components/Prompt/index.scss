/*=====================================
=            Prompt Styles            =
=====================================*/

@import '../../assets/styles/constants/layout';
@import '../../assets/styles/helpers/colors';
@import '../../assets/styles/helpers/units';

.prompt {
	z-index: 100;
	position: fixed;
	bottom: 0;
	left: 0;
	right: 0;
	display: inline-flex;
	justify-content: center;
	align-items: center;
	margin-bottom: -1px;
	transform: translateY(100%);
	transition: transform 250ms ease 500ms;
	user-select: none;
	pointer-events: none;
	&--is-visible {
		transform: translateY(0);
		transition-delay: 0s;
	}
	&--error,
	&--success {
		transition-delay: 0s;
		cursor: pointer;
	}
	&--error {
		.prompt__inner {
			background-color: get-color('red');
		}
	}
	&--success {
		background-color: get-color('red');
	}
	&__inner {
		max-width: 95%;
		min-height: rem(45);
		padding: rem(10) rem(25);
		border-top-right-radius: rem($border-radius);
		border-top-left-radius: rem($border-radius);
		background-color: get-color('gray-dark-extra');
		color: get-color('white');
		transition: background-color 250ms ease;
		pointer-events: auto;
	}
	&__spinner {
		& + * {
			margin-left: rem(10);
		}
	}
	&__text {
		font-size: rem(14);
		font-weight: 600;
		text-align: left;
	}
}
