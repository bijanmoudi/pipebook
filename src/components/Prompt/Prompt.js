import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Spinner } from '../../components';
import './index.scss';

const Prompt = ({
	text,
	visible,
	type,
	hasLoading,
	onClick,
	className,
	...restProps
}) => {
	return (
		<div
			role="alert"
			className={classnames(
				'prompt',
				{ [className]: className },
				{ [`prompt--${type}`]: type },
				{ [`prompt--is-visible`]: visible }
			)}
			{...restProps}
		>
			<div className="prompt__inner" onClick={() => onClick()}>
				{hasLoading && !['success', 'error'].includes(type) ? (
					<Spinner
						text={text}
						silent={!visible}
						className="prompt__spinner"
					/>
				) : null}
				<span className="prompt__text">{text}</span>
			</div>
		</div>
	);
};

Prompt.propTypes = {
	text: PropTypes.oneOfType([() => null, PropTypes.string.isRequired]),
	visible: PropTypes.bool,
	type: PropTypes.oneOfType([
		() => null,
		PropTypes.oneOf(['success', 'error', 'default'])
	]),
	hasLoading: PropTypes.bool,
	onClick: PropTypes.func
};

Prompt.defaultProps = {
	text: null,
	visible: false,
	type: null,
	hasLoading: false,
	onClick: () => {}
};

export default Prompt;
