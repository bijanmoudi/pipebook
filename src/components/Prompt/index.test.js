import React from 'react';
import renderer from 'react-test-renderer';
import Prompt from './Prompt';

const props = {
	text: 'An error occured',
	type: 'error',
	hasLoading: true
};

describe('Testing "Prompt" component', () => {
	it('Rendering loading Prompt without text', () => {
		const tree = renderer
			.create(<Prompt hasLoading={props.hasLoading} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Rendering failed Prompt', () => {
		const tree = renderer
			.create(<Prompt text={props.text} type={props.type} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
