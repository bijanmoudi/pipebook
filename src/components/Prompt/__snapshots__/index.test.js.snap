// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing "Prompt" component Rendering failed Prompt 1`] = `
<div
  className="prompt prompt--error"
  role="alert"
>
  <div
    className="prompt__inner"
    onClick={[Function]}
  >
    <span
      className="prompt__text"
    >
      An error occured
    </span>
  </div>
</div>
`;

exports[`Testing "Prompt" component Rendering loading Prompt without text 1`] = `
<div
  className="prompt"
  role="alert"
>
  <div
    className="prompt__inner"
    onClick={[Function]}
  >
    <div
      className="spinner prompt__spinner"
      style={Object {}}
    >
      <span
        aria-busy="true"
        className="visuallyhidden"
      />
    </div>
    <span
      className="prompt__text"
    />
  </div>
</div>
`;
